### PR TITLE
Allow rendering to be forced externally

### PR DESCRIPTION
**Pull request by hidmic**
# 🎉 New feature

## Summary
`ignition::gazebo::systems::Sensors` will only update its rendering sensors when there's a need for it. While perfectly reasonable, this also means that any custom rendering sensor relying on pre/post render hooks to do its work must wait for a builtin sensor that is known to this system to require an update. 

This patch works around this by introducing a `ForceRender` event that other systems can emit to force a rendering pass.

## Test it
...

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by hidmic** [comment]
Early push, to get feedback. I'll go implement a custom rendering sensor on top of this now.

File: include/ignition/gazebo/rendering/Events.hh [hunk 1]
```diff
@@ -65,6 +65,16 @@ namespace ignition
       /// \endcode
       using PostRender = ignition::common::EventT<void(void),
           struct PostRenderTag>;
+
+      /// \brief The force render event may be emitted outside the
+      /// rendering thread to force rendering calls.
```
Comment (chapulina) [inline include/ignition/gazebo/rendering/Events.hh:70]: The approach looks reasonable to me.

I'd just improve the documentation a bit to explain that "forcing" rendering means that when the time comes, a plugin that performs rendering will do so even if it wasn't planning to.

**Comment by hidmic** [comment]
A couple notes about tests and examples for this patch (which we're lacking). You can see in https://github.com/osrf/lrauv/pull/213 that putting together a plugin to make meaningful use of this new events is very tricky. So I think we have two choices: (a) we defer examples here and ticket a migration of https://github.com/osrf/lrauv/pull/213 into `gz-sensors` or (b) try to put together a dummy custom rendering sensor to motivate this changes. I'm biased towards (a), as I know (b) is going to be a rather long detour, full of booby traps.

**Comment by iche033** [comment]
The changes look fine. One note about going with this approach is that there is no guarantee that the custom rendering sensor's updates are lockstepped with physics. Something to keep in mind in case you are seeing that your sensor is not hitting the target update rate.

**Review by caguero — APPROVED** [review]
It looks reasonable to me.

**Comment by hidmic** [comment]
Alright, according to @caguero, failing checks are already happening on `master`. 

Moving forward!

**Pull request merged**